### PR TITLE
Fix warning in trivy scan and add error check to the command

### PR DESCRIPTION
**Pull request by joaopapereira**
#### What this PR does / why we need it:
Trivy scans are failing in strange error, in this PR we introduce the error checking with pipefail and also fix warnings of the action itself

##### Review Checklist:

- [ ] Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
- [ ] Relevant tests are added or updated
- [ ] Relevant docs in this repo added or updated
- [ ] Relevant carvel.dev docs added or updated in a separate PR and there's
  a link to that PR
- [ ] Code is at least as readable and maintainable as it was before this
  change

### CODE REVIEW

File: .github/workflows/trivy-scan.yml [hunk 1]
```diff
@@ -12,11 +12,12 @@ jobs:
...
         id: latest-sha
         run: |
+          set -eo pipefail
           # Get the latest released docker image sha
           curl -sL https://api.github.com/repos/vmware-tanzu/carvel-kapp-controller/releases/latest | jq -r '.assets[] | select(.name | contains("release.yml")).browser_download_url' | wget -i -
 
-          echo ::set-output name=image::$(yq eval '.spec.template.spec.containers[0].image' release.yml -N -oj | jq 'select(. != null)' -r)
-          echo ::set-output name=tag::$(curl -sL https://api.github.com/repos/vmware-tanzu/carvel-kapp-controller/releases/latest | jq -r '.tag_name')
+          echo "{image}={$(yq eval '.spec.template.spec.containers[0].image' release.yml -N -oj | jq 'select(. != null)' -r)}"  >> $GITHUB_OUTPUT
+          echo "{tag}={$(curl -sL https://api.github.com/repos/vmware-tanzu/carvel-kapp-controller/releases/latest | jq -r '.tag_name')}" >> $GITHUB_OUTPUT
```
Comment (praveenrewar) [inline .github/workflows/trivy-scan.yml:20]: Should we make this change in the other job ([here](https://github.com/vmware-tanzu/carvel-kapp-controller/blob/c5f626e3f6a85d689621ae25f4edb0e57d8c3e55/.github/workflows/trivy-scan.yml#L142)) as well?

**Review by praveenrewar — COMMENTED** [review]
LGTM!

**Comment by neil-hickey** [comment]
Rebase after https://github.com/vmware-tanzu/carvel-kapp-controller/pull/1033 is merged

**Review by neil-hickey — APPROVED** [review]
LGTM, thanks for fixing the warnings :)

**Pull request merged**